docs/tutorials/tutorial_2.rst

```diff
@@ -0,0 +1,537 @@
+Tutorial 2: Minimum Spanning Tree
+=================================
+
+In this tutorial, you will learn how to write a generator for a simple problem, called **Minimum Spanning Tree** with test groups. We will focus on the general idea of using test groups in tcframe.
+
+Here is the complete problem statement:
+
+----
+
+| **Minimum Spanning Tree**
+| Time Limit: 1 second
+| Memory Limit: 64 MB
+|
+| **Description**
+| Given a weighted connected graph G consisting of N nodes and M edges. The nodes are numbered from 0 to N - 1. You need to select N - 1 out of those M edges such that the graph remains connected if we only consider the selected edges. What is the minimum sum of the weight of the selected edges that satisfy the above constraint?
+|
+| **Input Format**
+| The first line contains two integers N and M separated by a space.
+| M lines follow. Each line contains three integers U, V, and W separated by a space. This line indicates that there is an edge connecting node U and V with weight W.
```

Comment:
Each line contains three integers U, V, and W separated by a space, representing an edge connecting node U and V with weight W.